crates/ironrdp-pdu/src/gcc/conference_create.rs

```diff
@@ -26,11 +26,26 @@ const CONFERENCE_NAME: &[u8] = b"1";
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct ConferenceCreateRequest {
-    pub gcc_blocks: ClientGccBlocks,
+    /// INVARIANT: `gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE <= u16::MAX`
+    gcc_blocks: ClientGccBlocks,
 }
 
 impl ConferenceCreateRequest {
     const NAME: &'static str = "ConferenceCreateRequest";
+
+    pub fn new(gcc_blocks: ClientGccBlocks) -> DecodeResult<Self> {
+        // INVARIANT: gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE <= u16::MAX
+        let _: usize = cast_length!(
+            "gcc blocks length",
+            gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE
+        )?;
```

Comment:
issue? Did you mean this?

```suggestion
        let _: u16 = cast_length!(
            "gcc blocks length",
            gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE
        )?;
```

Style-wise, see https://github.com/Devolutions/IronRDP/blob/master/STYLE.md#invariants

I would go for something like this:

```suggestion
        // Ensure the invariant on gcc_blocks.size() is respected.
        if !(gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE <= usize::from(u16::MAX)) {
            return Err(…);
        }
```

Only use `INVARIANT:` when defining the invariant (on the field), and reuse the same expression as-is, so it’s easy to verify.